Makefile

```diff
@@ -0,0 +1,109 @@
+#
+# Run static validations
+#
+# Terraform fmt
+terraform-fmt: ## Run command 'terraform fmt -check'
+	  docker run --rm -v $$PWD:/app -e AWS_ACCESS_KEY_ID=$$AWS_ACCESS_KEY_ID -e AWS_SECRET_ACCESS_KEY=$$AWS_SECRET_ACCESS_KEY hashicorp/terraform:$(TERRAFORM_VERSION) fmt -check
+# Terraform init
+terraform-init: ## Run command 'terraform init'
+	  docker run --rm -v $$PWD:/app -e AWS_ACCESS_KEY_ID=$$AWS_ACCESS_KEY_ID -e AWS_SECRET_ACCESS_KEY=$$AWS_SECRET_ACCESS_KEY hashicorp/terraform:$(TERRAFORM_VERSION) init
+# Terraform validate
+terraform-validate: ## Run command 'terraform validate'
+	  docker run --rm -v $$PWD:/app -e AWS_DEFAULT_REGION="us-east-1" -e AWS_ACCESS_KEY_ID=$$AWS_ACCESS_KEY_ID -e AWS_SECRET_ACCESS_KEY=$$AWS_SECRET_ACCESS_KEY hashicorp/terraform:$(TERRAFORM_VERSION) validate
+#
+# Run unit tests
+#
+unit-tests: terraform-fmt terraform-init terraform-validate ## Run unit tests (terraform fmt and validate)
+#
+# Run examples on a real AWS environment
+#
+# Private bucket
+#
+terraform-init-private-bucket: ## Private bucket example - Run command 'terraform init'
+	  $(call terraform-init,$(PRIVATE_EXAMPLE_DIR))
+
+terraform-plan-private-bucket: ## Private bucket example - Exec a terraform plan and puts it on a file called tfplan
+	  $(call terraform-plan,$(PRIVATE_EXAMPLE_DIR))
+
+terraform-apply-private-bucket: ## Private bucket example - Uses tfplan to apply the changes on AWS.
+	  $(call terraform-apply,$(PRIVATE_EXAMPLE_DIR))
+
+terraform-destroy-private-bucket: ## Private bucket example - Destroy all resources created by the terraform file in this repo.
+	  $(call terraform-destroy,$(PRIVATE_EXAMPLE_DIR))
+#
+# Public bucket
+#
+terraform-init-public-bucket: ## Public bucket example - Run command 'terraform init'
+	  $(call terraform-init,$(PUBLIC_EXAMPLE_DIR))
+
+terraform-plan-public-bucket: ## Public bucket example - Exec a terraform plan and puts it on a file called tfplan
+	  $(call terraform-plan,$(PUBLIC_EXAMPLE_DIR))
+
+terraform-apply-public-bucket: ## Public bucket example - Uses tfplan to apply the changes on AWS.
+	  $(call terraform-apply,$(PUBLIC_EXAMPLE_DIR))
+
+terraform-destroy-public-bucket: ## Public bucket example - Destroy all resources created by the terraform file in this repo.
+	  $(call terraform-destroy,$(PUBLIC_EXAMPLE_DIR))
+#
+# Run integration tests
+#
+# Localstack commands
+#
+localstack-start: ## Start localstack container
+		docker run -d --name $(LOCALSTACK_CONTAINER) --rm -p 4566:4566 -p 4571:4571 localstack/localstack -e "SERVICES=s3"
+
+localstack-stop: ## Stop localstack container
+		docker kill $(LOCALSTACK_CONTAINER)
+#
+# Private bucket
+#
+localtest-private-bucket: localstack-start ## Private bucket test - Run integration test on localstack
+		sleep 10
+		cd $(PRIVATE_TEST_DIR)/ && go test -v -timeout 30m
+		docker kill $(LOCALSTACK_CONTAINER)
+#
+# Public bucket
+#
+localtest-public-bucket: localstack-start ## Public bucket test - Run integration test on localstack
+		sleep 10
```

Comment:
Esse sleep é por qual motivo? Se for pra esperar o localstack, seria legal criar um código ou usar algum existente, pra fazer um loop e ficar tentar conectar na porta, assim ele fica tentando acessar a porta do localstack até que ele esteja pronto.

Reply:
Trataremos isso em um outro issue.